add module arg corresponding to cli refresh flag

smemsh opened the pull request:
This patch adds `-refresh=true` to the plan generation command line (which is the default if not specified), and adds a module argument `refresh` with default `true` value, so user can override and make sure resources are not refreshed for a specific module run.  This way, `plan` can be called with `-refresh=false` when a user knows they want this.

My use case: while implementing an image bake pipeline that uses Terraform to provision and Ansible to configure a node to be imaged, my play ends by cleaning out the cloud-init identity, shutting down the node, and then imaging the node.  Unfortunately, the state refresh that happens prior to imaging wakes up the node, and then it can't be imaged.  So I have a chicken-egg situation where I cannot plan the imaging step without waking up the node, whereas waking up the node prevents imaging.

Using `-refresh=false` during plan phase fixes this for my use case.  I think it's a valuable flag to expose to other users of the Ansible Terraform module.

Terraform source code has the first reference to `-refresh=` flag very early, in June 2014, so don't think we need a version check for this.

smemsh commented:
Rebased to current `main` tip after the recent release.  Can someone please look at this? Don't know who to ping, but I'll try @GomathiselviS maybe?

P.S. did not know what to use for `version_added`, will change it once someone lets me know.  Would it be perhaps 3.0.1 or 3.1.0 ?

GomathiselviS commented on plugins/modules/terraform.py:
```diff
@@ -129,6 +129,15 @@
+        value here.
+    type: bool
+    default: true
+    version_added: 1.0.0
```
```suggestion
    version_added: 3.2.0
```

smemsh commented:
So I rebased off the 4.0 `main` and changed the release fragment version to 4.0.1, however I'm a bit puzzled why 8 months has gone by and nobody commented on this PR or merged it? A major release came and went, others' PRs got merged, but nobody said anything about this one.  Is there something wrong with it? If it will never be merged, could you please let me know @GomathiselviS / @abikouo ? Thanks.
